### DISCUSSION TITLE
Installing on Ubuntu 20.04 from binaries alone

### DISCUSSION THREAD

**Discussion by CharlesJQuarra** [containers/podman#21382] [Q&A] [answered]
Since the podman version available for ubuntu 20.04 is ancient (3.4), and I can't be bothered to install 10Gb+ for a go toolchain and all dependencies, I downloaded the [binaries available here][3], as well as `gvproxy`:

    $ gvproxy --version
    gvproxy version 012bc90
    
```
$ runc --version
runc version 1.1.11
commit: v1.1.11-0-g4bccb38c
spec: 1.0.2-dev
go: go1.20.12
libseccomp: 2.5.4
```

```
$ conmon --version
conmon version 2.1.10
commit: 09bcded8e9c49cf1ff1fda403feac5a08f22535f-dirty
```

At this point, if I run

```
 $ podman info
    OS: linux/amd64
    provider: qemu
    version: 4.9.0
    
    Cannot connect to Podman. Please verify your connection to the Linux system using `podman system connection list`, or try `podman machine init` and `podman machine start` to manage a new Linux VM
    Error: unable to connect to Podman socket: failed to connect: dial tcp 127.0.0.1:41191: connect: connection refused
```

So, it seems something is missing, however I'm able to use a Qemu VM with `podman machine init` and `podman machine start`, this is the result afterwards of running `podman info`:

    host:
      arch: amd64
      buildahVersion: 1.33.2
      cgroupControllers:
      - cpu
      - io
      - memory
      - pids
      cgroupManager: systemd
      cgroupVersion: v2
      conmon:
        package: conmon-2.1.8-2.fc39.x86_64
        path: /usr/bin/conmon
        version: 'conmon version 2.1.8, commit: '
      cpuUtilization:
        idlePercent: 99.14
        systemPercent: 0.54
        userPercent: 0.32
      cpus: 4
      databaseBackend: sqlite
      distribution:
        distribution: fedora
        variant: coreos
        version: "39"
      eventLogger: journald
      freeLocks: 2043
      hostname: localhost.localdomain
      idMappings:
        gidmap:
        - container_id: 0
          host_id: 1000
          size: 1
        - container_id: 1
          host_id: 100000
          size: 1000000
        uidmap:
        - container_id: 0
          host_id: 1000
          size: 1
        - container_id: 1
          host_id: 100000
          size: 1000000
      kernel: 6.6.9-200.fc39.x86_64
      linkmode: dynamic
      logDriver: journald
      memFree: 1678553088
      memTotal: 2059902976
      networkBackend: netavark
      networkBackendInfo:
        backend: netavark
        dns:
          package: aardvark-dns-1.9.0-1.fc39.x86_64
          path: /usr/libexec/podman/aardvark-dns
          version: aardvark-dns 1.9.0
        package: netavark-1.9.0-1.fc39.x86_64
        path: /usr/libexec/podman/netavark
        version: netavark 1.9.0
      ociRuntime:
        name: crun
        package: crun-1.12-1.fc39.x86_64
        path: /usr/bin/crun
        version: |-
          crun version 1.12
          commit: ce429cb2e277d001c2179df1ac66a470f00802ae
          rundir: /run/user/1000/crun
          spec: 1.0.0
          +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +LIBKRUN +WASM:wasmedge +YAJL
      os: linux
      pasta:
        executable: /usr/bin/pasta
        package: passt-0^20231204.gb86afe3-1.fc39.x86_64
        version: |
          pasta 0^20231204.gb86afe3-1.fc39.x86_64
          Copyright Red Hat
          GNU General Public License, version 2 or later
            <https://www.gnu.org/licenses/old-licenses/gpl-2.0.html>
          This is free software: you are free to change and redistribute it.
          There is NO WARRANTY, to the extent permitted by law.
      remoteSocket:
        exists: true
        path: /run/user/1000/podman/podman.sock
      security:
        apparmorEnabled: false
        capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
        rootless: true
        seccompEnabled: true
        seccompProfilePath: /usr/share/containers/seccomp.json
        selinuxEnabled: true
      serviceIsRemote: true
      slirp4netns:
        executable: /usr/bin/slirp4netns
        package: slirp4netns-1.2.2-1.fc39.x86_64
        version: |-
          slirp4netns version 1.2.2
          commit: 0ee2d87523e906518d34a6b423271e4826f71faf
          libslirp: 4.7.0
          SLIRP_CONFIG_VERSION_MAX: 4
          libseccomp: 2.5.3
      swapFree: 0
      swapTotal: 0
      uptime: 0h 5m 0.00s
      variant: ""
    plugins:
      authorization: null
      log:
      - k8s-file
      - none
      - passthrough
      - journald
      network:
      - bridge
      - macvlan
      - ipvlan
      volume:
      - local
    registries:
      search:
      - docker.io
    store:
      configFile: /var/home/core/.config/containers/storage.conf
      containerStore:
        number: 5
        paused: 0
        running: 0
        stopped: 5
      graphDriverName: overlay
      graphOptions: {}
      graphRoot: /var/home/core/.local/share/containers/storage
      graphRootAllocated: 106769133568
      graphRootUsed: 4378746880
      graphStatus:
        Backing Filesystem: xfs
        Native Overlay Diff: "true"
        Supports d_type: "true"
        Supports shifting: "false"
        Supports volatile: "true"
        Using metacopy: "false"
      imageCopyTmpDir: /var/tmp
      imageStore:
        number: 52
      runRoot: /run/user/1000/containers
      transientStore: false
      volumePath: /var/home/core/.local/share/containers/storage/volumes
    version:
      APIVersion: 4.8.3
      Built: 1704291100
      BuiltTime: Wed Jan  3 09:11:40 2024
      GitCommit: ""
      GoVersion: go1.21.5
      Os: linux
      OsArch: linux/amd64
      Version: 4.8.3

I was under the impression that you didn't need a Qemu machine under linux, but it seems there should be something missing in my installation

The [podman installation guide](https://podman.io/docs/installation) gives no details on how to install from the provided binaries, so I just added them all to $PATH (which seems to have brought me up to 80% accomplishment)

  [1]: https://github.com/containers/podman/issues/20055
  [2]: https://github.com/containers/podman/issues/19639
  [3]: https://github.com/containers/podman/releases

**Comment by afbjorklund** [comment]
Only the `podman-remote` client binaries are provided by the release, to get the `podman` server you need to build from source.

Newer versions of Ubuntu have newer versions of Podman, but it is not backported: <https://packages.ubuntu.com/podman>

* https://github.com/containers/podman/discussions/17362

* https://github.com/containers/podman/issues/14302

**Comment by lstolcman** [comment]
There is kubic repo: https://podman.io/docs/installation#ubuntu

But due to updated version of go, which is not existing on older ubuntus, build is not working:
https://build.opensuse.org/package/show/devel:kubic:libcontainers:unstable/podman

I fixed the build script couple days ago, it [succeed](https://build.opensuse.org/package/show/home:lstolcman/podman) but it did not get merged yet: 
https://gitlab.com/rhcontainerbot/rpms-openqa/podman/-/merge_requests/5

I saw the kubic unstable is managed by @lsm5 but he posted a comment somewhere in the discussion that he dont have time to manage it.

@rhatdan @vrothberg @giuseppe  - could you advise how to push it forward?

I'm willing to work on fixing ubuntu build, as well as the ubuntu documentation (after build script got merged)

**Reply by afbjorklund** [reply]
I think the plan was to remove the "kubic" references, but maybe you could start a new OBS project for Ubuntu LTS?

* https://github.com/containers/podman.io/pull/240

**Reply by lsm5** [reply]
20.04 is really old and was never built on kubic-unstable. Maybe you wanna try forking the kubic repo and enabling 20.04 over there. If things work, we can look further.

**Comment by CharlesJQuarra** [comment, marked as answer]
FYI I was able to install successfully using this repo: https://github.com/mgoltzsche/podman-static 
```

$ podman info
host:
  arch: amd64
  buildahVersion: 1.33.3
  cgroupControllers: []
  cgroupManager: cgroupfs
  cgroupVersion: v1
  conmon:
    package: conmon_100:2.1.2~0_amd64
    path: /usr/libexec/podman/conmon
    version: 'conmon version 2.1.2, commit: '
  cpuUtilization:
    idlePercent: 78.99
    systemPercent: 3.27
    userPercent: 17.74
  cpus: 8
  databaseBackend: boltdb
  distribution:
    codename: focal
    distribution: pop
    version: "20.04"
  eventLogger: file
  freeLocks: 2048
  hostname: bygp-nova-01
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.11.0-7614-generic
  linkmode: dynamic
  logDriver: k8s-file
  memFree: 27559632896
  memTotal: 33620647936
  networkBackend: cni
  networkBackendInfo:
    backend: cni
    dns:
      package: podman-plugins_100:1.2.0-1_amd64
      path: /usr/libexec/cni/dnsname
      version: CNI dnsname plugin version unknown
    package: 'podman-machine-cni, podman-plugins: /usr/libexec/cni'
    path: /usr/libexec/cni
  ociRuntime:
    name: crun
    package: crun_100:1.2-2_amd64
    path: /usr/bin/crun
    version: |-
      crun version UNKNOWN
      commit: ea1fe3938eefa14eb707f1d22adff4db670645d6
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  pasta:
    executable: ""
    package: ""
    version: ""
  remoteSocket:
    exists: false
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/local/bin/slirp4netns
    package: Unknown
    version: |-
      slirp4netns version 1.2.2
      commit: 0ee2d87523e906518d34a6b423271e4826f71faf
      libslirp: 4.7.0
      SLIRP_CONFIG_VERSION_MAX: 4
      libseccomp: 2.5.4
  swapFree: 4294434816
  swapTotal: 4294434816
  uptime: 0h 3m 50.00s
  variant: ""
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - docker.io
  - registry.fedoraproject.org
  - registry.access.redhat.com
store:
  configFile: /home/charlesq/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.ignore_chown_errors: "true"
  graphRoot: /home/charlesq/.local/share/containers/storage
  graphRootAllocated: 462520029184
  graphRootUsed: 437365460992
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Supports shifting: "false"
    Supports volatile: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 2
  runRoot: /run/user/1000/containers
  transientStore: false
  volumePath: /home/charlesq/.local/share/containers/storage/volumes
version:
  APIVersion: 4.9.0
  Built: 0
  BuiltTime: Wed Dec 31 19:00:00 1969
  GitCommit: ""
  GoVersion: go1.20.13
  Os: linux
  OsArch: linux/amd64
  Version: 4.9.0

```

The only absolutely minor inconvenience is that I had already some work and images on the qemu VM, but I will remove that now and rebuild

thanks for the replies
